src/pt/mangotoons/src/eu/kanade/tachiyomi/extension/pt/mangotoons/MangoToons.kt

```diff
@@ -0,0 +1,206 @@
+package eu.kanade.tachiyomi.extension.pt.mangotoons
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import keiyoushi.utils.tryParse
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.Headers
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import uy.kohesive.injekt.injectLazy
+import java.text.DecimalFormat
+import java.text.DecimalFormatSymbols
+import java.text.SimpleDateFormat
+import java.util.Locale
+import java.util.TimeZone
+
+class MangoToons : HttpSource() {
+
+    override val name = "Mango Toons"
+
+    override val baseUrl = "https://mangotoons.com"
+
+    private val cdnUrl = "https://cdn.mangotoons.com"
+
+    override val lang = "pt-BR"
+
+    override val supportsLatest = true
+
+    override val client: OkHttpClient = network.cloudflareClient.newBuilder()
+        .rateLimit(2)
+        .build()
+
+    private val json: Json by injectLazy()
+
+    override fun headersBuilder(): Headers.Builder = super.headersBuilder()
+        .add("Referer", "$baseUrl/")
+        .add("Accept-Language", "pt-BR,pt;q=0.9,en-US;q=0.8,en;q=0.7")
+
+    // ================= Popular ===================
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$baseUrl/api/obras/top10/views?periodo=total", headers)
+    }
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val encrypted = response.body.string()
+        val decrypted = DecryptMango.decrypt(encrypted)
+
+        val result = json.decodeFromString<MangoResponse<List<MangoMangaDto>>>(decrypted)
+
+        val mangas = result.items.map { dto ->
+            SManga.create().apply {
+                title = dto.titulo
+                url = "/obra/${dto.slug ?: dto.id ?: dto.unique_id}"
+                thumbnail_url = (dto.capa ?: dto.imagem)?.let { "$cdnUrl/$it" }
```

Comment:
See if you can use https://kotlinlang.org/api/kotlinx.serialization/kotlinx-serialization-json/kotlinx.serialization.json/-json-names/ in the DTO instead of using fallbacks like this.